src/status_im2/subs/wallet/wallet.cljs

```diff
@@ -1,57 +1,53 @@
 (ns status-im2.subs.wallet.wallet
-  (:require [re-frame.core :as re-frame]
+  (:require [re-frame.core :as rf]
             [status-im2.contexts.wallet.common.utils :as utils]
             [utils.number]))
 
-(defn- calculate-raw-balance
-  [raw-balance decimals]
-  (if-let [n (utils.number/parse-int raw-balance nil)]
-    (/ n (Math/pow 10 (utils.number/parse-int decimals)))
-    0))
-
-(defn- total-per-token
-  [item]
-  (reduce (fn [ac balances]
-            (+ (calculate-raw-balance (:rawBalance balances)
-                                      (:decimals item))
-               ac))
-          0
-          (vals (:balancesPerChain item))))
+(rf/reg-sub
+ :wallet/ui
+ :<- [:wallet]
+ :-> :ui)
 
-(defn- calculate-balance
-  [address tokens]
-  (let [token  (get tokens (keyword address))
-        result (reduce
-                (fn [acc item]
-                  (let [total-values (* (total-per-token item)
-                                        (get-in item [:marketValuesPerCurrency :USD :price]))]
-                    (+ acc total-values)))
-                0
-                token)]
-    result))
+(rf/reg-sub
+ :wallet/tokens-loading?
+ :<- [:wallet/ui]
+ :-> :tokens-loading?)
 
-(re-frame/reg-sub
+(rf/reg-sub
  :wallet/accounts
  :<- [:wallet]
  :-> #(->> %
            :accounts
            vals
            (sort-by :position)))
 
-(re-frame/reg-sub
+(rf/reg-sub
  :wallet/balances
  :<- [:wallet/accounts]
- :<- [:wallet/tokens]
- (fn [[accounts tokens]]
-   (for [{:keys [address]} accounts]
-     {:address address
-      :balance (calculate-balance address tokens)})))
+ (fn [accounts]
+   (zipmap (map :address accounts)
+           (map #(-> % :tokens utils/calculate-balance) accounts))))
```

Comment:
balances is no longer a sequence of maps, now it's just a map. we no longer need a helper funciton, just extract the address from the map as any other regular key (i.e. `(get balances "0x1..")`)